### PR TITLE
insights: add & prepare to use generated store mocks

### PR DESCRIPTION
**Pull request by slimsag**
Helps #17218

### CODE REVIEW

File: enterprise/internal/insights/resolvers/insight_series_resolver_test.go [hunk 1]
```diff
@@ -21,19 +22,24 @@ func TestResolver_InsightSeries(t *testing.T) {
...
 		clock := func() time.Time { return now }
 		timescale, cleanup := insightsdbtesting.TimescaleDB(t)
-		defer cleanup()
 		postgres := dbtesting.GetDB(t)
 		resolver := newWithClock(timescale, postgres, clock)
 
+		// Create a mock store, delegating any un-mocked methods to the DB store.
+		dbStore := resolver.store
+		mockStore := store.NewMockInterfaceFrom(dbStore)
+		resolver.store = mockStore
```
Comment (slimsag) [inline enterprise/internal/insights/resolvers/insight_series_resolver_test.go:37]: Note: I plan to still use the actual DB store for some tests, to e.g. ensure that end-to-end querying for series data works as expected (integration tests)

**Comment by slimsag** [comment]
Note: This touched a lot of files because I updated the go-mockgen version we use. You can ignore the commit 0e32b64.

File: enterprise/internal/insights/store/interface.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package store
+
+import "context"
+
+//go:generate env GOBIN=$PWD/.bin GO111MODULE=on go install github.com/efritz/go-mockgen
```
Comment (efritz) [inline enterprise/internal/insights/store/interface.go:5]: I think we try to make gen.go the place for these (and nothing else), which can help with searchability of what triggers go generate.

File: go.mod [hunk 3]
```diff
@@ -36,7 +37,7 @@ require (
 	github.com/dnaeon/go-vcr v1.0.1
 	github.com/efritz/glock v0.0.0-20181228234553-f184d69dff2c
 	github.com/efritz/go-genlib v0.0.0-20200616012750-c21aae2e13ac // indirect
-	github.com/efritz/go-mockgen v0.0.0-20200524175724-37e2c732ee40
+	github.com/efritz/go-mockgen v0.0.0-20200916004441-cfcabc111002
```
Comment (efritz) [inline go.mod:40]: You fixed my speeling errors!

**Review by efritz — APPROVED** [review]

**Pull request merged**